File: 2014/10-17 PyCon PL Go/code/github-search.go
package main

import (
	"encoding/json"
	"fmt"
	"log"
	"net/http"
	"strings"
)

// typedefs OMIT

type GitHubSearchResponse struct {
	Items []*Repo
}

type Repo struct {
	Fullname string `json:"full_name"`
	Stars    int    `json:"stargazers_count"`
}

// -typedefs OMIT

func (sr *GitHubSearchResponse) print() {
	fmt.Printf(" #  %-20s %s\n", "Repo URL", "Stars")
	fmt.Println(strings.Repeat("-", 30))
	for i, r := range sr.Items {
		fmt.Printf("%02d. %-20s %d\n", i+1, r.Fullname, r.Stars)
	}
}

func main() {
	// Search GitHub repositories
	// req OMIT
	// massign OMIT
	query, language := "python", "go"
	// -massign OMIT
	url := `https://api.github.com/search/repositories?q=` + query +
		`+language:` + language + `&sort=stars&order=desc&per_page=3`
	resp, err := http.Get(url) // HLreq
	checkErr(err)
	defer resp.Body.Close() // HLdefer
	// -req OMIT

	// Decode JSON response
	// json OMIT
	dec := json.NewDecoder(resp.Body)
	var sr GitHubSearchResponse
	err = dec.Decode(&sr)
	checkErr(err)
	// -json OMIT

	// Print results
	sr.print()
}

func checkErr(err error) {
	if err != nil {
		log.Fatal("ERROR:", err)
		// fmt.Print(...)
		// os.Exit(1)
	}
}

File: 2014/10-17 PyCon PL Go/code/python/github-search.py
#!/usr/bin/env python
from __future__ import print_function
# // begin OMIT
import requests // HL

query, language = "python", "go"

url = ("https://api.github.com/search/repositories?q={}"
       "+language:{}&sort=stars&order=desc&per_page=3").format(query, language)
resp = requests.get(url) // HL
data = resp.json() // HL

print(" #  {:<20} {}".format("Repo URL", "Stars"))
print("-" * 30)
for i, r in enumerate(data["items"], 1):
    print("{:02d}. {:<20} {:d}".format(i, r["full_name"], r["stargazers_count"]))
